component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mobileconnect/MobileConnectAuthenticator.java

```diff
@@ -50,125 +46,484 @@
+                        // authorization endpoint
+                        authenticationEndpoint(context, response);
+                    }
+
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Error while retrieving properties. Authenticator Properties cannot be null");
+                    }
+                    throw new AuthenticationFailedException(" Authenticator Properties cannot be null");
+                }
+            }
+
+        }
+    }
+
+    /**
+     * Get the username of the logged in User.
+     */
+    private String getUsername(AuthenticationContext context) {
+        String username = null;
+        for (Integer stepMap : context.getSequenceConfig().getStepMap().keySet()) {
+            if (context.getSequenceConfig().getStepMap().get(stepMap).getAuthenticatedUser() != null
+                    && context.getSequenceConfig().getStepMap().get(stepMap).getAuthenticatedAutenticator()
+                    .getApplicationAuthenticator() instanceof LocalApplicationAuthenticator) {
+                username = String.valueOf(context.getSequenceConfig().getStepMap().get(stepMap).getAuthenticatedUser());
+                break;
             }
-            throw new AuthenticationFailedException(" Authenticator Properties cannot be null");
         }
+        return username;
+    }
+
+    /**
+     * Get the user realm of the logged in user.
+     */
+    private UserRealm getUserRealm(String username) throws AuthenticationFailedException {
+        UserRealm userRealm;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+            RealmService realmService = IdentityTenantUtil.getRealmService();
+            userRealm = (UserRealm) realmService.getTenantUserRealm(tenantId);
+        } catch (Exception e) {
+            throw new AuthenticationFailedException("Cannot find the user realm", e);
+        }
+        return userRealm;
+    }
 
+
+    /**
+     * Call the discovery endpoint with mcc and mnc.
+     */
+    private HttpURLConnection callDiscoveryWithMccMnc(HttpServletRequest request,
+                                                      Map<String, String> authenticatorProperties,
+                                                      String authorizationHeader) throws IOException {
+
+        //retrieve mcc_mnc from the request
+        String mccMncValue = request.getParameter(MobileConnectAuthenticatorConstants.
+                MOBILE_CONNECT_MCC_MNC);
+        //retrieve mcc and mnc from above string
+        String mcc = mccMncValue.substring(0, 3);
+        String mnc = mccMncValue.substring(4);
+
+        //call the discovery endpoint with the mcc and mnc
+        String url = MobileConnectAuthenticatorConstants.DISCOVERY_API_URL + "?" +
```

Comment:
Better encode. But this information being passed as parameters are not sensitive information. 

There is an instance where we pass the MSISDN of a user, but in a POST method. Does it get secured automatically in the POST body?